### DISCUSSION TITLE
[fixed] `ActiveSupport.on_load(:pay)` never runs

### DISCUSSION THREAD

**Discussion by marckohlbrugge** [pay-rails/pay#824] [Q&A] [answered]
**Update:** Fixed! I was still on `pay` `6.3.3` which doesn't support `on_load`.  I thought I already upgraded to the latest version, but the git branch I was working on was still behind 😅

---

I've been debugigng why my custom webhook subscribers don't work and finally narrowed it down to the `ActiveSupport.on_load(:pay)` block never running in my Rails app.

Here's my `config/initializers/pay.rb`:

```ruby
Pay.setup do |config|
  # *snip*
end

require Rails.root.join("app/models/billing.rb")
require Rails.root.join("app/models/billing/hooks.rb")
require Rails.root.join("app/models/billing/hooks/invoice_paid.rb")
require Rails.root.join("app/models/billing/hooks/customer_subscription_updated.rb")
require Rails.root.join("app/models/billing/hooks/customer_subscription_deleted.rb")

ActiveSupport.on_load(:pay) do
  Pay::Webhooks.delegator.subscribe "stripe.invoice.paid", Billing::Hooks::InvoicePaid.new
  Pay::Webhooks.delegator.subscribe "stripe.customer.subscription.updated", Billing::Hooks::CustomerSubscriptionUpdated.new
  Pay::Webhooks.delegator.subscribe "stripe.customer.subscription.deleted", Billing::Hooks::CustomerSubscriptionDeleted.new
end
```

With the above configuration, the `on_load` is never called and so the custom webhooks are not executed.

If I move the subscriptions outside of the `ActiveSupport.on_load(:pay)` block and just put them straight at the bottom of the initializer, then they _do_ get executed as expected.

Any idea why this might be the case? And is it safe not to use `on_load` ?

It's also worth pointing out that I need to manually require the my classes like you see in the example above, or I get the following error:

```
config/initializers/pay.rb:47:in <main>: uninitialized constant Billing (NameError)
```

A bit surprising, as I figured Rails/Zeitwerk typically takes care of loading files.

**Comment by excid3** [comment, marked as answer]
I was just going to ask if you were on the latest version. 

Also, Rails should autoload those files under app/models just fine so you shouldn't need to do any of those requires.
